synapse/storage/state.py

```diff
@@ -134,6 +136,23 @@ def from_lazy_load_member_list(members: Iterable[str]) -> "StateFilter":
             include_others=True,
         )
 
+    @staticmethod
+    def freeze(types: Mapping[str, Optional[Collection[str]]], include_others: bool):
```

Comment:
Somewhat feels like this could be replaced with `return StateFilter(synapse.util.frozenutil.freeze(types), include_others)`? I'm not sure if that's clearer or not though.